apps/designer/app/designer/features/style-panel/shared/css-value-input/unit-select.tsx

```diff
@@ -31,22 +37,29 @@ for (const [key, value] of unitRenderMap.entries()) {
 const defaultUnits = Array.from(unitRenderMap.keys());
 
 type UseUnitSelectType = {
+  property: string;
   value?: UnitValue;
   onChange: (value: StyleValue) => void;
   units?: Array<Unit>;
   onCloseAutoFocus: (event: Event) => void;
 };
 
 export const useUnitSelect = ({
+  property,
   onChange,
   value,
   units = defaultUnits,
   ...props
 }: UseUnitSelectType) => {
   const [isOpen, setIsOpen] = useState(false);
   const renderUnits = useMemo(
-    () => units.map((unit) => unitRenderMap.get(unit) ?? unit),
-    [units]
+    () =>
+      units
+        .filter((unit) => {
+          return isValid(property, toValue({ ...value, unit } as typeof value));
```

Comment:
the object passed to `toValue` has a property `value` that is of type: `number | undefined` but the expected type of that property is `string | number | string[]`. Here, `undefined` is not compatible with the expected type. The `undefined` appear here because the `value` identifier here has an optional value (its `value` property is `number` though, so it's compatible).

So for example this works OK:
```ts
          if (value) {
            toValue({ ...value, unit });
          }
```
But we need to also handle the other case somehow (when `value === undefined`). 

This one type checks OK:
```ts
  const renderUnits = useMemo(
    () =>
      value &&
      units
        .filter((unit) => {
          return isValid(property, toValue({ ...value, unit }));
        })
        .map((unit) => unitRenderMap.get(unit) ?? unit),
    [units, property, value]
  );
```

but it changes the type of `renderUnits` from `string[]` to `string[] | undefined` and this causes a problem later on.

In practice, this won't be a problem because we have an early return based on `value == undefined` but we can't express a relationship between `value` and `renderUnits` in a way that would allow TS to understand that checking `value` also checks~ `renderUnits` (unless we combine both into an object/array somehow).

The easiest fix is to change the latter check. So the next version could look smth like this:
```ts
  const renderUnits = useMemo(
    () =>
      value &&
      units
        .filter((unit) => {
          return isValid(property, toValue({ ...value, unit }));
        })
        .map((unit) => unitRenderMap.get(unit) ?? unit),
    [units, property, value]
  );

  const renderValue = value && unitRenderMap.get(value.unit);

  if (renderUnits == undefined || renderValue === undefined) {
    return [isOpen, null];
  }
```
That, in turn, will result in some further problems with the argument passed to `onChange`. So it seems that the easiest fix is to just check all of those things in the `if` responsible for the early return. The final version could look smth like this:
```ts
  const renderUnits = useMemo(
    () =>
      value &&
      units
        .filter((unit) => {
          return isValid(property, toValue({ ...value, unit }));
        })
        .map((unit) => unitRenderMap.get(unit) ?? unit),
    [units, property, value]
  );

  const renderValue = value && unitRenderMap.get(value.unit);

  if (value === undefined || renderUnits == undefined || renderValue === undefined) {
    return [isOpen, null];
  }
```

Is this extra added check somewhat redundant? Yes. Is this extra added check purely to satisfy the TS? Yes. 😉